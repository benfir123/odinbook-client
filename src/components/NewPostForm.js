import { Paper, Avatar, TextField } from "@mui/material";

const NewPostForm = ({
  user,
  postText,
  setPostText,
  postTextError,
  handlePostSubmit,
}) => {
  return (
    <Paper
      sx={{
        p: 2,
        display: "flex",
        flexDirection: "column",
        mb: 2,
      }}
    >
      <div style={{ display: "flex", alignItems: "center", gap: 15 }}>
        <Avatar alt={user.full_name} src={user.profile_pic_url} />
        <TextField
          placeholder={`What's on your mind, ${user.first_name}?`}
          variant="outlined"
          sx={{ flex: 1 }}
          onChange={(e) => setPostText(e.target.value)}
          value={postText}
          helperText={postTextError}
          error={postTextError ? true : false}
          onKeyDown={(e) => {
            if (e.key === "Enter") {
              handlePostSubmit(postText);
            }
          }}
        />
      </div>
    </Paper>
  );
};

export default NewPostForm;
