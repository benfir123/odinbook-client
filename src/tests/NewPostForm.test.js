import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import NewPostForm from "../components/NewPostForm";

describe("NewPostForm component", () => {
  it("renders the correct elements", () => {
    const { container } = render(
      <NewPostForm
        user={{
          full_name: "Ben Chanapai",
          first_name: "Ben",
          profile_pic_url: "",
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders the correct input value", () => {
    const onChangeMock = jest.fn();
    render(
      <NewPostForm
        user={{
          full_name: "Ben Chanapai",
          first_name: "Ben",
          profile_pic_url: "",
        }}
        setPostText={onChangeMock}
      />
    );
    const input = screen.getByRole("textbox");

    userEvent.click(input);
    userEvent.type(input, "Lion");

    expect(input).toHaveValue("Lion");
  });
});
