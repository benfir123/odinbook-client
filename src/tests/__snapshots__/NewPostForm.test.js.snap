// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPostForm component renders the correct elements 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-z6nvz2-MuiPaper-root"
  >
    <div
      style="display: flex; align-items: center; gap: 15px;"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
          data-testid="PersonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1vs1jgp-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            placeholder="What's on your mind, Ben?"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</div>
`;
